source/common/upstream/cluster_manager_impl.cc

```diff
@@ -419,9 +419,12 @@ void ClusterManagerImpl::createOrUpdateThreadLocalCluster(ClusterData& cluster)
               ENVOY_LOG(debug, "adding TLS cluster {}", new_cluster->name());
             }
 
-            cluster_manager.thread_local_clusters_[new_cluster->name()].reset(
-                new ThreadLocalClusterManagerImpl::ClusterEntry(cluster_manager, new_cluster,
-                                                                thread_aware_lb_factory));
+            auto thread_local_cluster = std::make_unique<ThreadLocalClusterManagerImpl::ClusterEntry>(
+                cluster_manager, new_cluster, thread_aware_lb_factory);
+            cluster_manager.thread_local_clusters_[new_cluster->name()].reset(thread_local_cluster.get());
```

Comment:
This code will double delete. You need to call `release()` here. The other way is to just store the allocation in a reference and then move it.

Reply:
Thank you for catching this; I was struggling to figure out why the integration tests were crashing.